FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc

```diff
@@ -224,11 +224,17 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle() {
       }
     }
 
-    // particle must not decay before it reaches the beam pipe
+    // FastSim will not make hits out of particles that decay before reaching the beam pipe
```

Comment:
Especially because this is FastSim, it is probably worth trying to avoid repeated computations of the same quantities.
A possibile suggestion:
```C++
    // particles which do not descend from exotics must be produced within the beampipe
    int exoticRelativeId = 0;
    bool producedWithinBeamPipe = productionVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_;
    if (!producedWithinBeamPipe) {
      exoticRelativesChecker(productionVertex, exoticRelativeId, 0);
      if (!isExotic(exoticRelativeId)) {
        continue;
      }
    }

    // FastSim will not make hits out of particles that decay before reaching the beam pipe
    bool decayedWithinBeamPipe = endVertex && endVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_;
    if (decayedWithinBeamPipe) {
      continue;
    }

    // SM particles that descend from exotics and cross the beam pipe radius should make hits but not be decayed
    if (producedWithinBeamPipe && !decayedWithinBeamPipe) {
      exoticRelativesChecker(productionVertex, exoticRelativeId, 0);
    }
```